### DISCUSSION TITLE
Module assets

### DISCUSSION THREAD

**Discussion by alphaCastor** [nuxt/nuxt#17230] [Questions]
Hi everyone. 
I have next module structure:

- src
  - runtime
    - assets/icons (.svg icons)
    - components
    - pages
  - module.ts
  
I managed to register components and pages by pages:extend and components:dirs hooks. 
And don't understand how can I use assets in playground. 
  
Any ideas?

**Comment by mirumirumi** [comment]
Hi!
Do you mean you want to be able to use each directory you list in any structure you want?
If so, `components` and `pages` are correct if you do as you did in nuxt.config.ts.

`assets` does not need to be registered with Nuxt in particular, as long as the directory is properly referenced when used in the application.

You can also use `alias` here:
```js
{
  "~~": "/<rootDir>",
  "@@": "/<rootDir>",
  "~": "/<rootDir>",
  "@": "/<rootDir>",
  "assets": "/<rootDir>/assets",     // <--- this!
  "public": "/<rootDir>/public"
}
```
see: https://v3.nuxtjs.org/api/configuration/nuxt.config#alias
